jfrog-cli/artifactory/commands/buildinfo/addgit.go

```diff
@@ -1,42 +1,312 @@
 	}
 
+	// Collect issues if required.
+	var issues []buildinfo.AffectedIssue
+	if config.ConfigFilePath != "" {
+		issues, err = config.collectBuildIssues()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Populate partials with VCS info.
 	populateFunc := func(partial *buildinfo.Partial) {
 		partial.Vcs = &buildinfo.Vcs{
 			Url:      gitManager.GetUrl(),
 			Revision: gitManager.GetRevision(),
 		}
+
+		if config.ConfigFilePath != "" {
+			partial.Issues = &buildinfo.Issues{
+				Tracker:                &buildinfo.Tracker{Name: config.IssuesConfig.TrackerName, Version: ""},
+				AggregateBuildIssues:   config.IssuesConfig.Aggregate,
+				AggregationBuildStatus: config.IssuesConfig.AggregationStatus,
+				AffectedIssues:         issues,
+			}
+		}
+	}
+	err = utils.SavePartialBuildInfo(config.BuildName, config.BuildNumber, populateFunc)
+	if err != nil {
+		return err
+	}
+
+	// Done.
+	log.Info("Collected git revision and remote url for", config.BuildName+"/"+config.BuildNumber+".")
+	return nil
+}
+
+func (config *BuildAddGitConfiguration) collectBuildIssues() ([]buildinfo.AffectedIssue, error) {
+	log.Info("Collecting build issues from VCS...")
+
+	// Initialize issues-configuration.
+	config.IssuesConfig = new(IssuesConfiguration)
+
+	// Build config from file.
+	err := config.IssuesConfig.populateIssuesConfigurations(config.ConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Run issues collection.
+	return config.doCollect(config.IssuesConfig)
+}
+
+func (config *BuildAddGitConfiguration) doCollect(issuesConfig *IssuesConfiguration) ([]buildinfo.AffectedIssue, error) {
+	// Create regex pattern.
+	issueRegexp, err := clientutils.GetRegExp(issuesConfig.Regexp)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create services manager to get build-info from Artifactory.
+	sm, err := utils.CreateServiceManager(issuesConfig.ArtDetails, false)
+	if err != nil {
+		return nil, err
+	}
+	// Get latest build-info from Artifactory.
+	buildInfoParams := services.BuildInfoParams{BuildName: config.BuildName, BuildNumber: config.BuildNumber}
+	buildInfo, err := sm.GetBuildInfo(buildInfoParams)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get previous VCS Revision from BuildInfo.
+	lastVcsRevision := ""
+	if buildInfo.Vcs != nil {
+		lastVcsRevision = buildInfo.Vcs.Revision
+	}
+
+	// Get log with limit, from latest commit.
+	logCmd := &LogCmd{gitPath: config.DotGitPath, logLimit: issuesConfig.LogLimit, lastVcsRevision: lastVcsRevision}
+	var foundIssues []buildinfo.AffectedIssue
+	protocolRegExp := gofrogcmd.CmdOutputPattern{
+		RegExp: issueRegexp,
+		ExecFunc: func(pattern *gofrogcmd.CmdOutputPattern) (string, error) {
+			// Reached here - means no error occurred.
+
+			// Check for out of bound results.
+			if len(pattern.MatchedResults) - 1 < issuesConfig.KeyGroupIndex || len(pattern.MatchedResults) - 1 < issuesConfig.SummaryGroupIndex {
+				return "", errors.New("Unexpected result while parsing build-issues from git log")
```

Comment:
build-issues --> issues
Let's also add to the above message - *Make sure that the regular expression used to find issues, includes two capturing groups, for the issue ID and the summary.*

Reply:
Fixed.